typed-racket-lib/typed-racket/rep/type-rep.rkt

```diff
@@ -124,7 +140,8 @@
                 [args exact-nonnegative-integer?]
                 [struct? boolean?])
   #:base
-  [#:custom-constructor
+  [#:custom-constructor/contract
```

Comment:
constructors can have a different interface than the underlying object (e.g. different number of arguments, different restrictions, etc)